garage/tf/models/base.py

```diff
@@ -0,0 +1,84 @@
+"""This file contains the abstraction class for models."""
+from garage.tf.models import AutoPickable
+
+
+class BaseModel:
```

Comment:
should this inherit from `keras.Model` (or `keras.Layer` or neither)?

@krzentner

Reply:
I think neither is slightly preferable. There's no particular reason to add the behavior of `tf.keras.Model` here.

Reply:
sorry i guess i was asking "to what extend does this interface overlap with tf.keras.Model"?